integration/tests/api/docker_test.go

```diff
@@ -270,3 +270,22 @@ func TestDockerContainerNetworkStats(t *testing.T) {
 	assert.NotEqual(stat.Network.RxBytes, stat.Network.TxBytes, "Network tx and rx bytes should not be equal")
 	assert.NotEqual(stat.Network.RxPackets, stat.Network.TxPackets, "Network tx and rx packets should not be equal")
 }
+
+// Tests goroutine leaks in docker handler.
+func TestGoRoutineSanity(t *testing.T) {
+	fm := framework.New(t)
+
+	assert := assert.New(t)
+	initialGoRoutines, err := fm.Cadvisor().GetGoRoutines()
+	assert.NoError(err)
+	containerId1 := fm.Docker().RunPause()
+	containerId2 := fm.Docker().RunPause()
+	waitForContainer(containerId1, fm)
+	waitForContainer(containerId2, fm)
+	fm.Cleanup()
+	// Adding the sleep here to let cAdvisor stabilize internally.
+	time.Sleep(5 * time.Second)
+	goRoutines, err := fm.Cadvisor().GetGoRoutines()
+	assert.NoError(err)
+	assert.True(goRoutines <= initialGoRoutines, "expected initial go routines to match current count.")
```

Comment:
Are we sure that there wouldn't be any short-lived goroutines causing the test to fail, e.g., goroutines serving http requests, etc? I am also not sure whether garbage collector may spawn goroutines, but I think it's possible. 

How about we start more containers in this test so that we can allow some margin for fluctuations in #goroutines.